lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -192,7 +192,7 @@ public function getDropIndexSQL($index, $table = null)
 
         return "IF EXISTS (SELECT * FROM sysobjects WHERE name = '$index')
```

Comment:
It seems that when a Table object is instantiated and hydrated from the schema-details, there are some conditions that cause indexes that don't actually exist (implicit indexes?) to be recorded on the Table object. This is ok in some cases, as the generated "to" schema usually has the implicit indexes as well.

One scenario where this messes up is when the primary key of an entity is also a foreign key to the id value of another entity, but the primary key that exists in the database includes a second column. Ideally, the schema diff should convert the existing primary key to only reference the single column, to match the entity. Instead, it tries to drop a nonexistent index with a generated name.

The same thing happens when a property flagged is flagged in the entity as a OneToOne relation. If a FK constraint already exists in the database, but no index exists for the column, the generated schema has a unique index for the column that does not match up with the implicit index generated when the existing schema is loaded. Thus, the schema diff tries to drop a nonexistent implicit index, and create a new unique index for the column. 

In the above scenario, it would be pretty much OK if the generated SQL would check if the nonexistent index exists before trying to delete it.